### ISSUE TITLE
Some services always apply certain events as true

### ISSUE THREAD

**Issue by Dridge** [egnyte/gitlabform#70]
When adjusting the Jira service the commit_events option is always true, and has to be set to false manually through the UI, the same is true for the mattermost service and the confidential_issues_events option.

```
project_settings:
    "group/subgroup/project":
      services:
          mattermost:
              active: true
              webhook: https://mattermost.com/hooks/xxx
              username: gitlab
              merge_requests_events: true
              merge_request_channel: "merge-requests"
              push_events: false
              issues_events: false
              confidential_issues_events: false # This doesnt appear to work, is always true, in all cases
              tag_push_events: false
              note_events: false
              confidential_note_events: false
              pipeline_events: false
              wiki_page_events: false
              branches_to_be_notified: "all"
          jira:
              active: true
              url: http://jiralo.exfo.com
              username: "user1"
              password: "password"
              merge_requests_events: true
              commit_events: false # This doesnt appear to work, is always true, in all cases
```

**Comment by gdubicki** [comment]
Sounds like another GitLab API bug... See also what I wrote in #66.

**Comment by gdubicki** [comment]
My new integration tests (see link above) confirm that there is an issue with JIRA service but not with Mattermost.

Can you verify with GitLab 12.6.2, @Dridge ?

**Comment by gdubicki** [comment]
While checking this I found out that there is a problem with setting `push_events` field using GitLab API too. You cannot set it to `false` when you create the service (its being created with it set to `true`) and when you have a service with it set to `true` you cannot change it to `false`. I reported a bug for this in GitLab here: https://gitlab.com/gitlab-org/gitlab/-/issues/209343

**Comment by gdubicki** [comment]
For the `commit_events` problem I have found an existing GitLab issue here: https://gitlab.com/gitlab-org/gitlab/-/issues/202216

**Comment by gdubicki** [comment]
Thanks to [Vincent Bisserié's help](https://gitlab.com/gitlab-org/gitlab/-/issues/202216#note_301934216) (if you are reading this - thank you!) I fixed the root cause and released v1.13.5 where this issue should be completely resolved without any workarounds.

Please reopen this issue if it does not help you, @Dridge .
